packages/react/src/utils/useWaitForComponentMount.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useRef, useState } from 'react';
+
+/**
+ * Used to detect when a Clerk component has been added to the DOM.
+ */
+function waitForElementChildren(options: { selector?: string; root?: HTMLElement | null; timeout?: number }) {
```

Comment:
This is somewhat of a brutce-force way of detecting when one of our components mounts. Instead of trying to plumb through some sort of event or callback to internal clerk-js components, I'm instead using a `MutationObserver` here to detect changes to the DOM. When the root element of the renderer has children (injected by clerk-js) we can assume the component has mounted.